firefox-ios/Client/Frontend/Autofill/Address/AddressListViewModel.swift

```diff
@@ -49,6 +49,10 @@ class AddressListViewModel: ObservableObject, FeatureFlaggable {
         let themeManager: ThemeManager = AppContainer.shared.resolve()
         return themeManager.getCurrentTheme(for: windowUUID).type == .dark
     }
+    var hasSyncableAccount: () -> Bool = {
+        let profile: Profile = AppContainer.shared.resolve()
```

Comment:
nit: we can add the profile inside the init and get it from there, wondering if you 

```
    init(profile: Profile = AppContainer.shared.resolve(),
```

Curious why we do it in var

Reply:
Hey Nish, so hasSyncableAccount can be overridden in tests just like `viewModel. hasSyncableAccount = { false }`. The reason I'm doing it like this is first this is the only place i really need profile, and second mocking it in tests would add additional overhead just to have hasSyncableAccount mocked.